dom.go

```diff
@@ -1946,120 +1992,196 @@ func (e *HTMLCanvasElement) GetContext(param string) *js.Object {
+
+func (ctx *CanvasRenderingContext2D) GetLineDash() interface{} {
+	return ctx.Call("getLineDash")
 }
 
-// Paths
+//
+func (ctx *CanvasRenderingContext2D) SetLineDash(arr interface{}) {
+	ctx.Call("setLineDash", arr)
+	return
+}
 
-func (ctx *CanvasRenderingContext2D) Fill() {
-	ctx.Call("fill")
+// Gradients and patterns
+
+func (ctx *CanvasRenderingContext2D) CreateLinearGradient(x0, y0, x1, y1 float64) {
+	ctx.Call("createLinearGradient", x0, y0, x1, y1)
 }
 
-func (ctx *CanvasRenderingContext2D) Stroke() {
-	ctx.Call("stroke")
+func (ctx *CanvasRenderingContext2D) CreateRadialGradient(x0, y0, r0, x1, y1, r1 float64) {
+	ctx.Call("createRadialGradient", x0, y0, r0, x1, y1, r1)
 }
 
-func (ctx *CanvasRenderingContext2D) BeginPath() {
-	ctx.Call("beginPath")
+func (ctx *CanvasRenderingContext2D) CreatePattern(image *Element, repetition string) {
+	ctx.Call("createPattern", image, repetition)
 }
 
-func (ctx *CanvasRenderingContext2D) MoveTo(x, y int) {
-	ctx.Call("moveTo", x, y)
+// Paths
+
+func (ctx *CanvasRenderingContext2D) BeginPath() {
+	ctx.Call("beginPath")
 }
 
 func (ctx *CanvasRenderingContext2D) ClosePath() {
 	ctx.Call("closePath")
 }
 
+func (ctx *CanvasRenderingContext2D) MoveTo(x, y int) {
+	ctx.Call("moveTo", x, y)
+}
+
 func (ctx *CanvasRenderingContext2D) LineTo(x, y int) {
 	ctx.Call("lineTo", x, y)
 }
 
-func (ctx *CanvasRenderingContext2D) Clip() {
-	ctx.Call("clip")
+func (ctx *CanvasRenderingContext2D) BezierCurveTo(cp1x, cp1y, cp2x, cp2y, x, y int) {
+	ctx.Call("bezierCurveTo", cp1x, cp1y, cp2x, cp2y, x, y)
 }
 
-func (ctx *CanvasRenderingContext2D) QuadraticCurveTo(cpx, cpy, x, y int) {
+func (ctx *CanvasRenderingContext2D) QuadraticCurveTo(cpx, cpy, x, y float64) {
 	ctx.Call("quadraticCurveTo", cpx, cpy, x, y)
 }
 
-func (ctx *CanvasRenderingContext2D) BezierCurveTo(cp1x, cp1y, cp2x, cp2y, x, y int) {
-	ctx.Call("bezierCurveTo", cp1x, cp1y, cp2x, cp2y, x, y)
-}
-
-func (ctx *CanvasRenderingContext2D) Arc(x, y, r int, sAngle, eAngle float64, counterclockwise bool) {
+func (ctx *CanvasRenderingContext2D) Arc(x, y, r, sAngle, eAngle float64, counterclockwise bool) {
 	ctx.Call("arc", x, y, r, sAngle, eAngle, counterclockwise)
 }
 
-func (ctx *CanvasRenderingContext2D) ArcTo(x1, y1, x2, y2, r int) {
+func (ctx *CanvasRenderingContext2D) ArcTo(x1, y1, x2, y2, r float64) {
 	ctx.Call("arcTo", x1, y1, x2, y2, r)
 }
 
-func (ctx *CanvasRenderingContext2D) IsPointInPath(x, y int) bool {
-	return ctx.Call("isPointInPath", x, y).Bool()
+func (ctx *CanvasRenderingContext2D) Ellipse(x, y, radiusX, radiusY, introtation, startAngle, endAngle float64, anticlockwise bool) {
+	ctx.Call("ellipse", x, y, radiusX, radiusY, introtation, startAngle, endAngle)
 }
 
-// Transformations
+func (ctx *CanvasRenderingContext2D) Rect(x, y, width, height float64) {
+	ctx.Call("rect", x, y, width, height)
+}
 
-func (ctx *CanvasRenderingContext2D) Scale(scaleWidth, scaleHeight int) {
-	ctx.Call("scale", scaleWidth, scaleHeight)
+//Drawing paths
```

Comment:
Add a space after `//` here for consistency.

```
// Drawing paths
```

Reply:
This isn't done yet.

Reply:
Fixed the typo.